### DISCUSSION TITLE
How to exclude subdirectories of docs from rerendering (quarto website hosted on github)?

### DISCUSSION THREAD

**Discussion by paulcbauer** [quarto-dev/quarto-web#1109] [Q&A]
I just built my website using quarto + github (the rendered website resides in the `docs` subfolder). I added some shinyapps that I add to subdirectories of `docs`, e.g., in a folder called `shinyapp_styling_markers`. I compile these shinyapps independently from the website using shinylive (e.g., https://paulcbauer.github.io/shinyapp_styling_markers/). The app sits in the same directory that contains the qmd files underlying the website in a subfolder folder called `2_shinyapp_styling_markers`.

The problem: Every time I rerender the quarto website all the subfolders of `docs` that I added independently are overwritten. Is there a way to avoid that, i.e., to exclude some `docs` subfolders from being overwritten once I render the website? I tried the below but it didn't work sofar. Thanks!

```
project:
  type: website
  output-dir: docs
  render:
    - "!shinyapp_styling_markers/"
```

**Comment by cderv** [comment]
For a Quarto project, output directory is always started french at each rendering. For anything that should be included in the output directory that is not rendered by quarto, it should be seen as resources. 

> I added some shinyapps that I add to subdirectories of docs, e.g., in a folder called shinyapp_styling_markers. I compile these shinyapps independently

Those subdirectories should be added a site resources: https://quarto.org/docs/websites/website-tools.html#site-resources
They will be copied into the output dir when quarto renders. 

Then if you need to ignore those folders from render target, you can either name the folder with a preprended `_` or using the render config as documented. https://quarto.org/docs/websites/#render-targets
Though for now there is a specificity as mentioned where you need to explicitly tells what you want to render before adding the ignored part. 

Hope it clarifies the behavior.

**Reply by paulcbauer** [reply]
Thank you that solved it. For anyone interested in my concrete approach. I keep all the shinyapps in a subfolder of the main directory "\_shinyapps_code". Because of "\_..." this folder is ignored when the website is being rendered. Then I compile my shiny apps into subfolders of my main directory which are then added as ressources (see below) and therefore not rendered but simply added asis in the output directory (here "docs") where the latter is recreated each time. My yaml below.

```
project:
  type: website
  output-dir: docs
  resources: 
    - "shinyapp_3d_layout"
    - "shinyapp_bullseye"
    - "shinyapp_plotly_layout"
    - "shinyapp_styling_markers"
  render:
    - "*.qmd" # _shinyapps_code is not rendered because of "_"
``
